src/cclib/io/moldenwriter.py

```diff
@@ -10,19 +10,28 @@
 import os.path
 
 from . import filewriter
+from .filewriter import MissingAttributeError
 from cclib.parser import utils
 
 
 class MOLDEN(filewriter.Writer):
     """A writer for MOLDEN files."""
 
+    def _check_required_attributes(self, required):
```

Comment:
I was thinking this should go in the base ```Writer``` class, and then the required attributes could be passed into the constructor:

```
def __init__(self, ccdata, jobfilename=None, terse=False,
                 required_attrs=[], *args, **kwargs):
```

You'll have to make sure that this doesn't break anything downstream though.